### PR TITLE
Add support for building singularity/apptainer containers 

### PR DESCRIPTION
**Pull request by nurbal**
This PR is about adding support of singularity in the "container" command added by PR #17. Therefore, it should not be merged before #17.

### CODE REVIEW

**Comment by breuleux** [comment]
I merged #17. Is this ready or still a WIP?

**Comment by nurbal** [comment]
Still a wip. I will do my best to finish this asap.

File: milabench/cli.py [hunk 1]
```diff
@@ -319,4 +329,34 @@ def dockerfile_template(milabench_req, include_data, use_conda, python_version):
...
+    mkdir /bench
+    mkdir /base
+    echo '{ milabench_req }' > /version.txt
+    pip install -U pip && \
+    pip install -r /version.txt && \
+    milabench install /bench/bench.yaml && \
+    rm -rf /root/.cache/pip
+
+%startscript
+{ '    milabench prepare /bench/bench.yaml' if include_data else '' }
```
Comment (abergeron) [inline milabench/cli.py:357]: This should be in post.  The idea is to have the dataset as part of the image if `include_data` is True

File: milabench/cli.py [hunk 2]
```diff
@@ -288,7 +289,16 @@ def container():
...
+                with (root / "milabench.def").open("w") as f:
+                    f.write(
+                        singularitydef_template(
+                            milabench_req="git+https://github.com/mila-iqia/milabench.git@container",
+                            include_data=include_data,
+                            use_conda=use_conda,
+                            python_version=python_version,
+                        )
+                    )
+                subprocess.check_call(["sudo", "singularity", "build", tag+".sif", "milabench.def"], cwd=root)
```
Comment (abergeron) [inline milabench/cli.py:301]: You should copy the output_dir stuff above.

**Comment by breuleux** [comment]
@nurbal I'm taking over the PR, since you're on vacation. Thanks!

**Pull request merged**